src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -158,7 +241,19 @@ private void PrepareForVerify(ReadOnlySpan<byte> content, out int alg, out byte[
             }
 
             alg = nullableAlg.Value;
-            toBeSigned = CreateToBeSigned(SigStructureCoxtextSign1, _protectedHeaderAsBstr, content);
+
+            byte[] rentedbuffer = ArrayPool<byte>.Shared.Rent(ComputeToBeSignedEncodedSize(SigStructureCoxtextSign1, _protectedHeaderAsBstr, content));
+
+            try
+            {
+                Span<byte> buffer = rentedbuffer;
+                int bytesWritten = CreateToBeSigned(SigStructureCoxtextSign1, _protectedHeaderAsBstr, content, buffer);
```

Comment:
In the event someone does something like

```C#
foreach (RSA key in knownKeysRsa)
{
    if (msg.Verify(key))
    {
        return true;
    }
}

// Repeat with ECDSA

return false;
```

It might be worth adding `byte[]? _toBeSigned` as a field, and this method returns the field if it's non-null, or computes and assigns it otherwise.

Reply:
If someone calls `msg.Verify(key, Encoding.UTF8.GetBytes("foo"))` and then `msg.Verify(key, Encoding.UTF8.GetBytes("bar"))`, then the cached TBS will be in a bad state, so I don't think we should do this (I already did it in the latest commit but will revert it and add a test).

Reply:
Or we can cache it only in the case where content is embedded?

Reply:
Ah, yes, detached content.  If you don't think it's worth doing for just the attached/embedded case that's fine, it was merely a passing thought.